#161967087: Users can signup/register to Authors Haven

Bryan-Cee opened the pull request:
**What does this PR do?**
-  Enable users to create an account.

**Description of tasks to be completed?**
- Create an account
- Send activation link to user via email


**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Switch to the appropriate branch with `gco ft-signup-register-161967087`
- Run `npm install` to update your packages.
- Fire up the server with `npm start`
- Navigate to `http://localhost:3000/signup`

**What are the relevant pivotal tracker stories?**
- [#161967087](https://www.pivotaltracker.com/story/show/161967087)

**Screenshots**
<img width="926" alt="screenshot 2019-01-17 at 17 48 17" src="https://user-images.githubusercontent.com/19145546/51326501-52df1b00-1a80-11e9-94b1-da01b9363fb5.png">
<img width="971" alt="screenshot 2019-01-17 at 17 49 17" src="https://user-images.githubusercontent.com/19145546/51326761-e7e21400-1a80-11e9-8ae6-98df9504cc78.png">

Bryan-Cee committed:
feat(sign-up): users can signup/register

 - Enable user to sign up
 - Return consise responses
 - Send email to user with activation link

[Delivers #161967087]

bettblake08 commented on src/actions/RegistrationActions.js:
```diff
@@ -0,0 +1,55 @@
+
+const registrationAsync = user => (dispatch) => {
+  dispatch(registerRequest());
+  instance
```
Can you move the API call to an API class method, then dispatch actions according to what has been returned from API class method?

bettblake08 commented on src/components/app.js:
```diff
@@ -1,62 +1,25 @@
-    actions.welcomeAction(welcomeMessage);
-    ArticleAPI.getArticles();
-  }
+  componentWillMount() {}
```
I believe you should remove this. This lifecycle method is deprecated.

bettblake08 commented on src/reducers/index.js:
```diff
@@ -1,8 +1,9 @@
 export default history => combineReducers({
   router: connectRouter(history),
-  welcomeReducer
+  registration
```
Add the suffix Reducer to the registration reducer.

Bryan-Cee committed:
refactor: refactor components, add feed back

Bryan-Cee committed:
refactor: renamed registrationAction module

Bryan-Cee committed:
test(adding tests): add test for actions

bettblake08 reviewed (approved):
👍

Darius-Ndubi reviewed (approved):
👍

Bryan-Cee committed:
refactor: rename reducer in registration container

justmesam commented on src/actions/registrationActions.js:
```diff
@@ -0,0 +1,55 @@
+/* eslint-disable import/prefer-default-export */
+export const registerRequest = () => ({
+  type: REGISTER_REQUEST,
+  isFetching: true
```
you don't have to have this as part of the action  you only need to change it on the reducer

justmesam commented on src/assets/style/main.scss:
```diff
@@ -1,6 +1,20 @@
+}
+
+.ui.active.transition.visible.dimmer,
+.ui.page.modals.dimmer.transition.visible.active {
```
which class is this and wouldn't have been easier to create a new reference class ?

Bryan-Cee replied:
Both classes are inaccessible directly, they are generated by semantic UI React components, that is why I chose to pick the whole class as is to effect the change that was needed.

justmesam commented on src/assets/style/main.scss:
```diff
@@ -1,6 +1,20 @@
+}
+
+.modal {
+  text-align: right !important;
```
refrain from using important.

Bryan-Cee replied:
I needed to override the default text-align property set by semantic UI React components, is there any other way to have it done without using `! important` ?

justmesam replied:
you can unset the value

justmesam commented on src/components/containers/registration.js:
```diff
@@ -0,0 +1,131 @@
+
+          if (value.match(usernameReg) === null) {
+            this.setState({
+              ...this.state,
```
why are you spreading the state here??

Bryan-Cee committed:
refactor: implement feedback

 - remove spreading state in this.setState
   in registration.js
 - refactored actions
 - updated reducer to handle the actions
   properly
 - refactored action tests

[Delivers #161967087 ]
